[Bug] - Having anything else than en-US first in Accepted-Language headers kill the client

### Requirements

- [X] This is a bug report, and if not, please post to https://lemmy.ml/c/lemmy_support instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this issue already exists.
- [X] It's a single bug. Do not report multiple bugs in one issue.
- [X] It's a frontend issue, not a backend issue; Otherwise please create an issue on the [backend repo](https://github.com/LemmyNet/lemmy) instead.

### Summary

Seems that the bug returns.

Now, if you setup your browser as accepting, for example, french first so that it sends the header `Accepted-Language: fr-FR,en-US;q=0.7,en;q=0.3`, nothing displays in the UI.

The client does not look for the right path to resolve the reference to the local js file.

```
Uncaught (in promise) Error: Cannot find module './fr-FR'
    a https://jlai.lu/static/9e7fec7/js/client.js:2
    promise callback*a https://jlai.lu/static/9e7fec7/js/client.js:2
    e https://jlai.lu/static/9e7fec7/js/client.js:2
    p https://jlai.lu/static/9e7fec7/js/client.js:2
    C https://jlai.lu/static/9e7fec7/js/client.js:2
    w https://jlai.lu/static/9e7fec7/js/client.js:2
    e https://jlai.lu/static/9e7fec7/js/client.js:2
    s https://jlai.lu/static/9e7fec7/js/client.js:2
    t https://jlai.lu/static/9e7fec7/js/client.js:2
    t https://jlai.lu/static/9e7fec7/js/client.js:2
    Ki https://jlai.lu/static/9e7fec7/js/client.js:2
    Zi https://jlai.lu/static/9e7fec7/js/client.js:2
    e https://jlai.lu/static/9e7fec7/js/client.js:2
    p https://jlai.lu/static/9e7fec7/js/client.js:2
    C https://jlai.lu/static/9e7fec7/js/client.js:2
    w https://jlai.lu/static/9e7fec7/js/client.js:2
    e https://jlai.lu/static/9e7fec7/js/client.js:2
    s https://jlai.lu/static/9e7fec7/js/client.js:2
    t https://jlai.lu/static/9e7fec7/js/client.js:2
    t https://jlai.lu/static/9e7fec7/js/client.js:2
    Kf https://jlai.lu/static/9e7fec7/js/client.js:2
    <anonymous> https://jlai.lu/static/9e7fec7/js/client.js:2
    <anonymous> https://jlai.lu/static/9e7fec7/js/client.js:2
    <anonymous> https://jlai.lu/static/9e7fec7/js/client.js:2
    <anonymous> https://jlai.lu/static/9e7fec7/js/client.js:2
```

It should have at least a gracious fallback mechanism since this is not the first time it breaks.

### Steps to Reproduce

1. Open your browser and set its accepted language as French, for example on Firefox:
![image](https://github.com/LemmyNet/lemmy-ui/assets/72401951/45cb1132-7a3d-46c9-9aaf-f78867d64ae8)

Then 

![image](https://github.com/LemmyNet/lemmy-ui/assets/72401951/b70c4d56-ad60-410e-99e3-6bbe051e417b)
2. Load a lemmy instance
3. Look:

![image](https://github.com/LemmyNet/lemmy-ui/assets/72401951/5f2de80d-29bc-4e87-bdbb-9313f3daa527)

4. If you still see content, check in your dev tools that the request is sent with the  header value starting with fr-FR `Accepted-Language: fr-FR,en-US;q=0.7,en;q=0.3`,



### Technical Details

Seems that the client looks for the translation file in the wrong directory again or the file is not named as expected.

```
Uncaught (in promise) Error: Cannot find module './fr-FR'
    a https://jlai.lu/static/9e7fec7/js/client.js:2
    promise callback*a https://jlai.lu/static/9e7fec7/js/client.js:2
    e https://jlai.lu/static/9e7fec7/js/client.js:2
    p https://jlai.lu/static/9e7fec7/js/client.js:2
    C https://jlai.lu/static/9e7fec7/js/client.js:2
    w https://jlai.lu/static/9e7fec7/js/client.js:2
    e https://jlai.lu/static/9e7fec7/js/client.js:2
    s https://jlai.lu/static/9e7fec7/js/client.js:2
    t https://jlai.lu/static/9e7fec7/js/client.js:2
    t https://jlai.lu/static/9e7fec7/js/client.js:2
    Ki https://jlai.lu/static/9e7fec7/js/client.js:2
    Zi https://jlai.lu/static/9e7fec7/js/client.js:2
    e https://jlai.lu/static/9e7fec7/js/client.js:2
    p https://jlai.lu/static/9e7fec7/js/client.js:2
    C https://jlai.lu/static/9e7fec7/js/client.js:2
    w https://jlai.lu/static/9e7fec7/js/client.js:2
    e https://jlai.lu/static/9e7fec7/js/client.js:2
    s https://jlai.lu/static/9e7fec7/js/client.js:2
    t https://jlai.lu/static/9e7fec7/js/client.js:2
    t https://jlai.lu/static/9e7fec7/js/client.js:2
    Kf https://jlai.lu/static/9e7fec7/js/client.js:2
    <anonymous> https://jlai.lu/static/9e7fec7/js/client.js:2
    <anonymous> https://jlai.lu/static/9e7fec7/js/client.js:2
    <anonymous> https://jlai.lu/static/9e7fec7/js/client.js:2
    <anonymous> https://jlai.lu/static/9e7fec7/js/client.js:2
```

### Lemmy Instance Version

0.18.1-rc5

### Lemmy Instance URL

https://jlai.lu

--- a/src/shared/utils/app/setup-date-fns.ts
+++ b/src/shared/utils/app/setup-date-fns.ts
@@ -7,6 +7,16 @@ export default async function () {
     lang = "en-US";
   }
 
+  // if lang and country are the same, then date-fns expects only the lang
+  // eg: instead of "fr-FR", we should import just "fr"
+
+  if (lang.includes("-")) {
+    const parts = lang.split("-");
+    if (parts[0] === parts[1].toLowerCase()) {
+      lang = parts[0];
+    }
+  }
+
   const locale = (
     await import(
       /* webpackExclude: /\.js\.flow$/ */

